arbor/affinity.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <string>
+
+#ifdef ARB_HAVE_HWLOC
+#include <hwloc.h>
+#endif
+
+#include <arbor/util/scope_exit.hpp>
+
+namespace arb {
+
+enum class affinity_kind {
+  thread,
+  process
+};
+
+
+#ifdef ARB_HAVE_HWLOC
+
+inline
+void hwloc(int exp, const std::string& msg) {
+    if (-1 == exp) {
+        throw arbor_internal_error(std::string{"HWLOC Thread failed at: "} + msg);
```

Comment:
It's a mysterious error with a mysterious error message. I would prefer to take responsibility for it: what is the consequence of the error? Maybe sth like: "Arbor cannot proceed, due to an error in HWLOC: %msg. Consider doing X/changing Y."

Reply:
It's a mystery wrapped in an enigma.

Reply:
Merci.